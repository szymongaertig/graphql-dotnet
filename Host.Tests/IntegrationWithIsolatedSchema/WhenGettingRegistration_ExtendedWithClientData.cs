using Host.Registrations;
using Host.Tests.Mocks;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Xunit;

namespace Host.Tests.Units;

public class WhenGettingRegistration_ExtendedWithClientData
{
    [Fact]
    public async Task ForNotExistingRegistration_ThenReturnsValidRegistrationId()
    {
        // Arrange
        var fixture = new RegistrationFixture();
        var dbContext = fixture.GetDbContext();
        await dbContext.Database.MigrateAsync();
        var registration = new RegistrationMockBuilder().Build();

        var request = fixture.BuildQueryRequest(@"query getRegistrations($registrationId: Int!) {
          registrations(where: {id : {eq: $registrationId }}) {
            nodes {
              id
            }
          }
        }", new Dictionary<string, object>()
        {
            { "registrationId", registration.Id }
        });

        var requestExecutor = await new ServiceCollection()
            .AddDbContextPool<RegistrationDbContext>(optionsBuilder =>
                optionsBuilder.UseSqlite(fixture.GetConnectionString()))
            .AddGraphQLServer()
            .AddRegistartionGraphQL()
            .BuildRequestExecutorAsync();

        // Act
        var result = await requestExecutor.ExecuteAsync(request);

        // Assert
        result.ExpectQueryResult();
        var jsonResult = result.ExpectQueryResult().ToDictionary();
    }
}