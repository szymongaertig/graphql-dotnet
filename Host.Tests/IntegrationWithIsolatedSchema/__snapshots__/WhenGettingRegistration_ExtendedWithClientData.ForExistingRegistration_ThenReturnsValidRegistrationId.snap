{
  "Kind": "SingleResult",
  "Label": null,
  "Path": null,
  "Data": {
    "registrations": {
      "nodes": []
    }
  },
  "Items": null,
  "Errors": null,
  "Extensions": null,
  "Incremental": null,
  "ContextData": null,
  "HasNext": null
}
